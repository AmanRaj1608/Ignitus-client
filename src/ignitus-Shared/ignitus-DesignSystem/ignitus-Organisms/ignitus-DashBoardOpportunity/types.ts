export type Opportunity = {
  title: string;
  description: string;
  subtitle: string;
  src: string;
};
