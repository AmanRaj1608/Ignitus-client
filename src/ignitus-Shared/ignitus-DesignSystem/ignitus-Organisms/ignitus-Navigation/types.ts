export interface NavigationProps {
  transparentNavigation?: boolean;
  padding?: string;
}

export type ToggleProps = {
  isExpanded: boolean;
};

export type NavigationItems = {
  title: string;
  route: string;
};
