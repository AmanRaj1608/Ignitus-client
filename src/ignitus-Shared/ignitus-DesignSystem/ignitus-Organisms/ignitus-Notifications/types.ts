export type InputTypes = {
  colorText: string;
  colorTitle: string;
  background: string;
  image?: string;
  title?: string;
  textLeft?: string;
  textRight?: string;
};

export type StyledTagProps = {
  background: string;
  color: string;
};
