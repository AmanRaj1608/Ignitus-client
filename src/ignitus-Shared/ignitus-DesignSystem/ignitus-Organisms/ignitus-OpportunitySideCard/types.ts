export type Opportunity = {
  title: string;
  description: string;
};

export type Cardtypes = {
  location: string;
  commitment: string;
  salary: string;
};
