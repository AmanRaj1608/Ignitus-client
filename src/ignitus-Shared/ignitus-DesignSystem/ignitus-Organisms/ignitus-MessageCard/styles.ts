import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import * as T from '../../ignitus-Atoms/typography';
import { flexibleRowDiv } from '../../shared';
import { Normal } from '../../ignitus-Atoms/fonts';

export const MessageContainer = styled(flexibleRowDiv)`
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  padding: 1rem;
  width: 50%;
`;

export const ImageContainer = styled(flexibleRowDiv)`
  padding-right: 1rem;
  img {
    position: relative;
    border-radius: 50%;
    width: 56px;
    max-width: 100%;
    max-height: 100%;
  }
  svg {
    color: #219653;
    position: absolute;
    stroke: white;
    fill: #219653;
  }
`;

export const MessageSubTitle = styled(T.Paragraph)`
  padding-right: 1rem;
  font-weight: ${Normal};
  color: ${C.GreyOne};
`;

export const MessageDateTitle = styled.div`
  text-align: right;
`;
