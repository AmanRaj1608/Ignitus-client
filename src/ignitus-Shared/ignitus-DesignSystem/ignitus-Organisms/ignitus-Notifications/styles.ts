import styled from '@emotion/styled';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { StyledTagProps } from './types';

export const StyledTag = styled.span<StyledTagProps>`
  background: ${props => props.background};
  color: ${props => props.color};
  text-align: center;
  border-radius: 2rem;
  line-height: 1rem;
  padding: 1rem 2rem;
  display: inline-flex;
  align-items: center;
  margin: 0.5rem;
  cursor: pointer;
  width: cover;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const IconContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
`;

export const Image = styled.img`
  height: 3rem;
  width: 3rem;
  border-radius: 50%;
  margin-right: 1rem;
`;

export const Text = styled.span`
  font-weight: bold;
  color: ${props => props.color};
`;

export const LeftContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
`;

export const RightContainer = styled.div``;
