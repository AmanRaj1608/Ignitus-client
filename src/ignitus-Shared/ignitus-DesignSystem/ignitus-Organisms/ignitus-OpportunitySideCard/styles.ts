import styled from '@emotion/styled';
import { White, GreyBackground, IgnitusBlue } from '../../ignitus-Atoms/colors';
import { SemiBold } from '../../ignitus-Atoms/fonts';
import { Heading4 } from '../../ignitus-Atoms/typography';
import { RoundedButton } from '../../ignitus-Atoms/buttons';
import I from '../../../ignitus-Utilities/Components/icon';

export const Container = styled.div`
  border-radius: 1rem;
  width: 26rem;
  background-color: ${White};
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: hidden;
  max-height: 35rem;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 20rem;
  padding: 3rem;
  border-bottom: 0.2rem solid ${GreyBackground};
  overflow: hidden;
`;

export const ButtonsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 25rem;
  padding: 2rem;
`;

export const Heading = styled(Heading4)`
  font-weight: ${SemiBold};
`;

export const Button = styled(RoundedButton)`
  width: 11rem;
`;

export const Icon = styled(I)`
  fill: ${IgnitusBlue};
`;
