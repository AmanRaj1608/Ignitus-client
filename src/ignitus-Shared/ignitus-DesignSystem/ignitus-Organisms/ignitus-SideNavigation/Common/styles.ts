import styled from '@emotion/styled';

import { White, IgnitusBlue, GreyLight } from '../../../ignitus-Atoms/colors';
import { MD, SemiBold, Normal } from '../../../ignitus-Atoms/fonts';
import { StyledLayerProps } from './types';

export const Layer = styled.div<StyledLayerProps>`
  border-bottom: ${props => (props.border ? `1px solid ${GreyLight}` : 'null')};
  color: ${IgnitusBlue};
  cursor: pointer;
  font-size: ${MD};
  font-weight: ${props => (props.semiBold ? `${SemiBold}` : `${Normal}`)};
  padding: 1rem 0;
  width: 100%;
  .NavDropdown {
    display: flex;
    align-items: center;
    justify-content: center;
  }
  &:hover {
    background-color: ${IgnitusBlue};
    color: ${White};
    p {
      color: ${White};
    }
    ${props =>
      props.containIcon
        ? `
        svg {
          fill: ${White};
        }`
        : ''}
  }
`;
