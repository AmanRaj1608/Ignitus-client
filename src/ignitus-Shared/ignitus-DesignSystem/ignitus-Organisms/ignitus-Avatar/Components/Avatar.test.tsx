import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { Avatar } from './Avatar';
import { AvatarImage } from '../styles';
import { avatar } from '../types';

const pre = 'https://storage.googleapis.com/ignitus_assets/ig-avatars/';

const avatarNames = ['grant', 'melanie', 'george', 'eugene'];

const randomAvatar = () =>
  avatarNames[Math.floor(Math.random() * avatarNames.length)];

const props: avatar = {
  name: randomAvatar(),
  src: `${pre}${randomAvatar()}.png`,
};

let wrapper;

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

describe('<Avatar />', () => {
  beforeEach(() => {
    wrapper = mount(<Avatar />);
  });

  it('should render', () => {
    wrapper = render(<Avatar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Avatar images', () => {
    expect(wrapper.find(AvatarImage).length).toBe(1);

    wrapper = mount(<AvatarImage {...props} />);

    expect(wrapper.find(AvatarImage).prop('src')).toEqual(props.src);
    expect(wrapper.find(AvatarImage).prop('name')).toEqual(props.name);
  });
});
