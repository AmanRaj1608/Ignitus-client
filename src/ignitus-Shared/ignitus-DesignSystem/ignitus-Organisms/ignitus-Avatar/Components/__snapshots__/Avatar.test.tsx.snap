// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render 1`] = `
.emotion-0 {
  max-width: 9rem;
}

<img
  alt="melanie"
  class="emotion-0"
  src="https://storage.googleapis.com/ignitus_assets/ig-avatars/melanie.png"
/>
`;
